### PR TITLE
Minor changes in topology generator

### PR DESCRIPTION
**Pull request by mskd12**
- Update README
- Fix pep8 issues
  <a href='#crh-start'></a><a href='#crh-data-%7B%22processed%22%3A%20%5B%22https%3A//github.com/netsec-ethz/scion/pull/232%23issuecomment-118283015%22%2C%20%22https%3A//github.com/netsec-ethz/scion/pull/232%23issuecomment-118283068%22%2C%20%22https%3A//github.com/netsec-ethz/scion/pull/232%23issuecomment-118283405%22%2C%20%22https%3A//github.com/netsec-ethz/scion/pull/232%23issuecomment-118284958%22%2C%20%22https%3A//github.com/netsec-ethz/scion/pull/232%23issuecomment-118365973%22%2C%20%22https%3A//github.com/netsec-ethz/scion/pull/232%23discussion_r33849838%22%2C%20%22https%3A//github.com/netsec-ethz/scion/pull/232%23discussion_r33849941%22%2C%20%22https%3A//github.com/netsec-ethz/scion/pull/232%23discussion_r33850004%22%2C%20%22https%3A//github.com/netsec-ethz/scion/pull/232%23discussion_r33850604%22%2C%20%22https%3A//github.com/netsec-ethz/scion/pull/232%23discussion_r33851182%22%2C%20%22https%3A//github.com/netsec-ethz/scion/pull/232%23discussion_r33868300%22%2C%20%22https%3A//github.com/netsec-ethz/scion/pull/232%23discussion_r33868301%22%2C%20%22https%3A//github.com/netsec-ethz/scion/pull/232%23discussion_r33868302%22%2C%20%22https%3A//github.com/netsec-ethz/scion/pull/232%23discussion_r34129417%22%2C%20%22https%3A//github.com/netsec-ethz/scion/pull/232%23discussion_r34129714%22%2C%20%22https%3A//github.com/netsec-ethz/scion/pull/232%23discussion_r34141730%22%5D%2C%20%22comments%22%3A%20%7B%22General%20Comment%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/netsec-ethz/scion/pull/232%23issuecomment-118283015%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22Since%20you%20are%20fixing%20this%2C%20here%27s%20a%20few%20other%20suggestions%3A%5Cr%5Cn%28New%29%20Line%20128%3A%20add%20some%20indentation%20before%20%60isd_graph.degree%28...%60%22%2C%20%22created_at%22%3A%20%222015-07-03T09%3A11%3A22Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/1120468%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/rev112%22%7D%7D%2C%20%7B%22body%22%3A%20%22Line%20342%3A%20spaces%20around%20%60%2A%60%22%2C%20%22created_at%22%3A%20%222015-07-03T09%3A11%3A42Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/1120468%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/rev112%22%7D%7D%2C%20%7B%22body%22%3A%20%22PEP%208%20naming%20conventions%3A%20it%27s%20better%20to%20rename%20%60ISD_dict%60%2C%20%60ISD_NUM%60%2C%20%60NUM_CORE_ADS%60.%20Make%20them%20lowercase%2C%20for%20example.%22%2C%20%22created_at%22%3A%20%222015-07-03T09%3A13%3A03Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/1120468%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/rev112%22%7D%7D%2C%20%7B%22body%22%3A%20%22Lines%2096%2C%20118%3A%20%60isd%60%20in%20lambda%20expression%20shadows%20%60isd%60%20from%20outer%20scope%22%2C%20%22created_at%22%3A%20%222015-07-03T09%3A15%3A03Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/1120468%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/rev112%22%7D%7D%2C%20%7B%22body%22%3A%20%22%3A%2B1%3A%3Ca%20href%3D%27%23crh-update-none%27%3E%3C/a%3E%22%2C%20%22created_at%22%3A%20%222015-07-03T14%3A42%3A46Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/1120468%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/rev112%22%7D%7D%5D%2C%20%22title%22%3A%20%22General%20Comment%22%7D%2C%20%22Pull%20c23bb6606576519eb2daae9db87fe4c955578d39%20topology/brite/topology_generator.py%20103%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/netsec-ethz/scion/pull/232%23discussion_r33849838%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22There%27s%20another%20unfixed%20comparison%20with%20None%20at%20line%20312%20%28319%29.%22%2C%20%22created_at%22%3A%20%222015-07-03T09%3A04%3A59Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/1120468%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/rev112%22%7D%7D%2C%20%7B%22body%22%3A%20%22%3A%2B1%3A%3Ca%20href%3D%27%23crh-update-none%27%3E%3C/a%3E%22%2C%20%22created_at%22%3A%20%222015-07-03T14%3A42%3A46Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/1120468%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/rev112%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20topology/brite/topology_generator.py%3AL379-392%22%7D%2C%20%22Pull%20c23bb6606576519eb2daae9db87fe4c955578d39%20topology/brite/README.md%2018%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/netsec-ethz/scion/pull/232%23discussion_r33850604%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22I%20think%20it%20also%20makes%20sense%20to%20add%20a%20note%20about%20requirements%3A%20at%20this%20point%2C%20%60networkx%60%20should%20be%20installed%20additionally.%20Not%20sure%20if%20you%20should%20add%20it%20to%20global%20%60requirements.txt%60%2C%20though.%20%40kormat%3F%22%2C%20%22created_at%22%3A%20%222015-07-03T09%3A19%3A47Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/1120468%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/rev112%22%7D%7D%2C%20%7B%22body%22%3A%20%22Keeping%20the%20amount%20of%20dependencies%20low%20is%20good%2C%20but%20%60networkx%60%20looks%20to%20be%20fairly%20small%2C%20and%20doesn%27t%20pull%20in%20a%20lot%20of%20other%20dependencies.%20Why%20%60argparse%60%2C%20though%3F%20We%20are%20standardised%20on%20python3.4%2C%20which%20ships%20with%20argparse%20included.%22%2C%20%22created_at%22%3A%20%222015-07-08T09%3A19%3A42Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/6919822%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/kormat%22%7D%7D%2C%20%7B%22body%22%3A%20%22I%20will%20add%20%60networkx%60%20to%20the%20requirements.txt%20then%22%2C%20%22created_at%22%3A%20%222015-07-08T09%3A23%3A24Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/6560106%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/mskd96%22%7D%7D%2C%20%7B%22body%22%3A%20%22%3A%2B1%3A%20%22%2C%20%22created_at%22%3A%20%222015-07-08T12%3A31%3A23Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/6919822%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/kormat%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20topology/brite/README.md%3AL21-37%22%7D%2C%20%22Pull%20c23bb6606576519eb2daae9db87fe4c955578d39%20topology/brite/topology_generator.py%2052%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/netsec-ethz/scion/pull/232%23discussion_r33849941%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22Redundant%20parentheses%22%2C%20%22created_at%22%3A%20%222015-07-03T09%3A07%3A01Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/1120468%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/rev112%22%7D%7D%2C%20%7B%22body%22%3A%20%22%3A%2B1%3A%3Ca%20href%3D%27%23crh-update-none%27%3E%3C/a%3E%22%2C%20%22created_at%22%3A%20%222015-07-03T14%3A42%3A46Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/1120468%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/rev112%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20topology/brite/topology_generator.py%3AL249-256%22%7D%2C%20%22Pull%20c23bb6606576519eb2daae9db87fe4c955578d39%20topology/brite/topology_generator.py%2094%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/netsec-ethz/scion/pull/232%23discussion_r33850004%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22Wait%2C%20is%20%60list_labels%60%20a%20list%20of%20lambdas%20now%3F%22%2C%20%22created_at%22%3A%20%222015-07-03T09%3A08%3A14Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/1120468%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/rev112%22%7D%7D%2C%20%7B%22body%22%3A%20%22Oh..%20I%20wanted%20to%20fix%20some%20pep8%20warning%20and%20I%20introduced%20a%20bug..%20I%20will%20revert%20it%20back%22%2C%20%22created_at%22%3A%20%222015-07-03T09%3A29%3A51Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/6560106%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/mskd96%22%7D%7D%2C%20%7B%22body%22%3A%20%22%3A%2B1%3A%3Ca%20href%3D%27%23crh-update-complete%27%3E%3C/a%3E%22%2C%20%22created_at%22%3A%20%222015-07-03T14%3A42%3A46Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/1120468%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/rev112%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20topology/brite/topology_generator.py%3AL359-367%22%7D%7D%7D'></a>
  <a href='https://www.codereviewhub.com/'></a>
- [x] <a href='#crh-comment-Pull c23bb6606576519eb2daae9db87fe4c955578d39 topology/brite/topology_generator.py 103'></a> &nbsp;<b><a href='https://github.com/netsec-ethz/scion/pull/232#discussion_r33849838'>File: topology/brite/topology_generator.py:L379-392</a></b>
- <a href='https://github.com/rev112'></a> There's another unfixed comparison with None at line 312 (319).
- [x] <a href='#crh-comment-Pull c23bb6606576519eb2daae9db87fe4c955578d39 topology/brite/topology_generator.py 52'></a> &nbsp;<b><a href='https://github.com/netsec-ethz/scion/pull/232#discussion_r33849941'>File: topology/brite/topology_generator.py:L249-256</a></b>
- <a href='https://github.com/rev112'></a> Redundant parentheses
- [x] <a href='#crh-comment-Pull c23bb6606576519eb2daae9db87fe4c955578d39 topology/brite/topology_generator.py 94'></a> &nbsp;<b><a href='https://github.com/netsec-ethz/scion/pull/232#discussion_r33850004'>File: topology/brite/topology_generator.py:L359-367</a></b>
- <a href='https://github.com/rev112'></a> Wait, is `list_labels` a list of lambdas now?
- <a href='https://github.com/mskd96'></a> Oh.. I wanted to fix some pep8 warning and I introduced a bug.. I will revert it back
- [x] <a href='#crh-comment-General Comment'></a> &nbsp;<b><a href='https://github.com/netsec-ethz/scion/pull/232#issuecomment-118283015'>General Comment</a></b>
- <a href='https://github.com/rev112'></a> Since you are fixing this, here's a few other suggestions:
  (New) Line 128: add some indentation before `isd_graph.degree(...`
- <a href='https://github.com/rev112'></a> Line 342: spaces around `*`
- <a href='https://github.com/rev112'></a> PEP 8 naming conventions: it's better to rename `ISD_dict`, `ISD_NUM`, `NUM_CORE_ADS`. Make them lowercase, for example.
- <a href='https://github.com/rev112'></a> Lines 96, 118: `isd` in lambda expression shadows `isd` from outer scope
- [x] <a href='#crh-comment-Pull c23bb6606576519eb2daae9db87fe4c955578d39 topology/brite/README.md 18'></a> &nbsp;<b><a href='https://github.com/netsec-ethz/scion/pull/232#discussion_r33850604'>File: topology/brite/README.md:L21-37</a></b>
- <a href='https://github.com/rev112'></a> I think it also makes sense to add a note about requirements: at this point, `networkx` should be installed additionally. Not sure if you should add it to global `requirements.txt`, though. @kormat?
- <a href='https://github.com/kormat'></a> Keeping the amount of dependencies low is good, but `networkx` looks to be fairly small, and doesn't pull in a lot of other dependencies. Why `argparse`, though? We are standardised on python3.4, which ships with argparse included.
- <a href='https://github.com/mskd96'></a> I will add `networkx` to the requirements.txt then

<a href='https://www.codereviewhub.com/netsec-ethz/scion/pull/232?mark_as_completed=1'></a>&nbsp;<a href='https://www.codereviewhub.com/netsec-ethz/scion/pull/232?approve=1'></a>&nbsp;<a href='https://github.com/netsec-ethz/scion/pull/232'></a>
<a href='#crh-end'></a>

### CODE REVIEW

File: topology/brite/topology_generator.py [hunk 1]
```diff
@@ -371,12 +379,13 @@ def json_convert(graph):
             if links[isd_ad_id_neighbor] == "PARENT":
                 cert_issuer = isd_ad_id_neighbor
         topo_dict[isd_ad_id]["links"] = links
-        if cert_issuer != None:
+        if cert_issuer is not None:
```
Comment (tonyo) [inline topology/brite/topology_generator.py:384]: There's another unfixed comparison with None at line 312 (319).

File: topology/brite/topology_generator.py [hunk 2]
```diff
@@ -244,6 +249,7 @@ def read_topo_file(topo_file, ISD_NUM):
             edges_count = int(values[2])
     return (original_graph, num_outedges)
```
Comment (tonyo) [inline topology/brite/topology_generator.py]: Redundant parentheses

File: topology/brite/topology_generator.py [hunk 3]
```diff
@@ -351,8 +359,8 @@ def json_convert(graph):
     topo_dict = dict()
     topo_dict["default_subnet"] = "127.0.0.0/8"
     for isd_ad_id in graph.nodes():
-        func_labels = lambda x: graph.edge[isd_ad_id][x]['label']
-        list_labels = [func_labels(x) for x in list(graph.edge[isd_ad_id])]
+        list_labels = [lambda x: graph.edge[isd_ad_id][x]['label'](x)
+                       for x in list(graph.edge[isd_ad_id])]
```
Comment (tonyo) [inline topology/brite/topology_generator.py]: Wait, is `list_labels` a list of lambdas now?

File: topology/brite/README.md [hunk 4]
```diff
@@ -21,4 +21,16 @@ Necessary steps to use the topology generator
...
+
+All command options for running topology_generator.py
+
+	a) -f (or) --file: Give all brite files seperately one after another
+
+	b) -d (or) --dir: Convert all files in the specified directory. Each of the file is converted into an ISD, then core AD's in these ISD's are connected
+
+	c) -c (or) --degree: Two arguments to be given - min and max degrees for connections between core AD's. It is used for generating inter-ISD connections
+
+	d) -o (or) --output: Generate a dot output file
```
Comment (tonyo) [inline topology/brite/README.md:38]: I think it also makes sense to add a note about requirements: at this point, `networkx` should be installed additionally. Not sure if you should add it to global `requirements.txt`, though. @kormat?

File: topology/brite/topology_generator.py (hunk 3 above)
Comment (mskd12) [inline topology/brite/topology_generator.py]: Oh.. I wanted to fix some pep8 warning and I introduced a bug.. I will revert it back

File: topology/brite/topology_generator.py (hunk 1 above)
Comment (tonyo) [inline topology/brite/topology_generator.py:384]: :+1:<a href='#crh-update-none'></a>

File: topology/brite/topology_generator.py (hunk 2 above)
Comment (tonyo) [inline topology/brite/topology_generator.py]: :+1:<a href='#crh-update-none'></a>

File: topology/brite/topology_generator.py (hunk 3 above)
Comment (tonyo) [inline topology/brite/topology_generator.py]: :+1:<a href='#crh-update-complete'></a>

File: topology/brite/README.md (hunk 4 above)
Comment (kormat) [inline topology/brite/README.md:38]: Keeping the amount of dependencies low is good, but `networkx` looks to be fairly small, and doesn't pull in a lot of other dependencies. Why `argparse`, though? We are standardised on python3.4, which ships with argparse included.

Comment (mskd12) [inline topology/brite/README.md:38]: I will add `networkx` to the requirements.txt then

Comment (kormat) [inline topology/brite/README.md:38]: :+1:

**Pull request merged**